src/Expander.hs

```diff
@@ -226,11 +226,189 @@ visit modName = do
               getState
   unless visitedp $ do
     let m' = shift i m -- Shift the syntax literals in the module source code
-    evalResults <- inPhase p $ evalMod m'
+    sc <- freshScope $ T.pack $ "For module-phase " ++ shortShow modName ++ "-" ++ shortShow p
+    let syntaxScopeSets :: Traversal' Syntax ScopeSet
```

Comment:
This seems generally useful to me - perhaps it should live in Syntax.hs?

Reply:
even better, I've managed to replace these very long Traversal definitions with a generic `allScopeSets` which can produce a `Traversal` of all the `ScopeSet`s in any type which has a `Data` instance.